lib/src/merge.rs

```diff
@@ -816,6 +1030,45 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_update_from_simplified() {
+        // 1-way merge
+        assert_eq!(
+            c(&[], &[0]).update_from_simplified(&c(&[], &[1])),
+            c(&[], &[1])
+        );
+        // 3-way merge
+        assert_eq!(
+            c(&[0], &[0, 0]).update_from_simplified(&c(&[], &[1])),
+            c(&[0], &[0, 1])
+        );
+        assert_eq!(
+            c(&[0], &[1, 0]).update_from_simplified(&c(&[], &[0])),
```

Comment:
Perhaps replace the last 0 with a 2, so that the result is [0], [2,0]? Or you could add a test like that.